lib/views/frontend/spree/checkout/payment/_stripe.html.erb

```diff
@@ -1,71 +1,57 @@
+  let elements = stripe.elements();
+  let card = elements.create('card', {
+    style: {
+      base: {
+        lineHeight: "20px"
+      }
+    }
+  });
 
-  var mapCC, stripeResponseHandler;
 
-  mapCC = function(ccType) {
-    if (ccType === 'MasterCard') {
-      return 'mastercard';
-    } else if (ccType === 'Visa') {
-      return 'visa';
-    } else if (ccType === 'American Express') {
-      return 'amex';
-    } else if (ccType === 'Discover') {
-      return 'discover';
-    } else if (ccType === 'Diners Club') {
-      return 'dinersclub';
-    } else if (ccType === 'JCB') {
-      return 'jcb';
-    }
-  };
+  function stripeTokenHandler(token) {
+    // Insert the token ID into the form so it gets submitted to the server
+    let form = $(Spree.stripePaymentMethod.parents("form").get(0));
+    let paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
 
-  stripeResponseHandler = function(status, response) {
-    var paymentMethodId, token;
-    if (response.error) {
-      $('#stripeError').html(response.error.message);
-      return $('#stripeError').show();
-    } else {
-      Spree.stripePaymentMethod.find('#card_number, #card_expiry, #card_code').prop("disabled", true);
-      Spree.stripePaymentMethod.find(".ccType").prop("disabled", false);
-      Spree.stripePaymentMethod.find(".ccType").val(mapCC(response.card.brand));
-      token = response['id'];
-      paymentMethodId = Spree.stripePaymentMethod.prop('id').split("_")[2];
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][gateway_payment_profile_id]' value='" + token + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][last_digits]' value='" + response.card.last4 + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][month]' value='" + response.card.exp_month + "'/>");
-      Spree.stripePaymentMethod.append("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][year]' value='" + response.card.exp_year + "'/>");
-      return Spree.stripePaymentMethod.parents("form").get(0).submit();
-    }
+    form.append($("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][gateway_payment_profile_id]' value='" + token.id + "'/>"));
+    form.append($("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][last_digits]' value='" + token.card.last4 + "'/>"));
+    form.append($("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][month]' value='" + token.card.exp_month + "'/>"));
+    form.append($("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][year]' value='" + token.card.exp_year + "'/>"));
```

Comment:
I don't think `class` is needed?

A helper method would make this more readable and remove code duplication:
```
  function addInput(name, value) {
    form.append($("<input type='hidden' class='stripeToken' name='payment_source[" + paymentMethodId + "][" + name + "]' value='" + value + "'/>"));
  }

  addInput("gateway_payment_profile_id",  token.id);
  addInput("last_digits",  token.card.last4);
  addInput("month",  token.card.exp_month);
  addInput("year",  token.card.exp_year);
```